### PR TITLE
Backwards compatibility for process_output handling empty linter results

### PR DESCRIPTION
**Pull request by jayvdb**
Depends on https://github.com/coala/coala-bears/pull/2658 which in turn depends on https://github.com/coala/coala-bears/pull/2758

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,33 +2,75 @@
...
+
 python_paths = .ci/
 
+reqsfilenamepatterns =
+  requirements.txt
+  test-requirements.txt
+
+reqsignorevcs = True
+
+timeout = 35
```
Comment (li-boxuan) [inline setup.cfg]: appveyor is complaining about timeout problem... Maybe also increase the timeout value in this PR?

Comment (jayvdb) [reply setup.cfg]: That commit isnt part of this PR.

I saw the problem once or twice, at least once in DocGrammarBear.  Needs a separate issue.  As that is a native bear, it should be quick -- if it isnt likely the problem is in the bear logic.

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
Codecov diff https://codecov.io/gh/coala/coala-bears/pull/2761/diff showing coverage of `if True` branch

File: bears/js/JSComplexityBear.py [hunk 2]
```diff
@@ -34,7 +34,10 @@ def process_output(self, output, filename, file,
         :param cc_threshold: Threshold value for cyclomatic complexity
         """
         message = '{} has a cyclomatic complexity of {}.'
-        if output:
+        if not output:  # backwards compatible no results
+            return
+
+        if True:  # no else
```
Comment (li-boxuan) [inline bears/js/JSComplexityBear.py]: `# pragma: no branch`?

Comment (jayvdb) [reply bears/js/JSComplexityBear.py]: Could be used, but we'll also get rid of that soon.  This is a more specific rule.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-bears/pull/2761/commits/3502e945c5cbd6dc2a15de7965ea653159119647 https://github.com/coala/coala-bears/pull/2761/commits/1fafcbe24232c6fec4cb6dc8ca40ec66ad31dcd8 https://github.com/coala/coala-bears/pull/2761/commits/f8695413d9eaf225799fc38471af62b75ca21c3d https://github.com/coala/coala-bears/pull/2761/commits/2cba79a602cc2a9d023f0fd9bbc68f61b5e88296

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**